src/org/labkey/test/tests/ProjectCreatorUserTest.java

```diff
@@ -0,0 +1,138 @@
+public class ProjectCreatorUserTest extends BaseWebDriverTest
+{
+    private static final String PROJECT_CREATOR_USER = "project_creator@permission.test";
+    private static final String READER = "reader@permission.test";
+    private static String PROJECT_NAME_PC = "Folder by Project Creator";
+
+    @BeforeClass
+    public static void setup()
+    {
+        ProjectCreatorUserTest initTest = (ProjectCreatorUserTest) getCurrentTest();
+        initTest.doSetup();
+    }
+
+    @Override
+    protected @Nullable String getProjectName()
+    {
+        return getClass().getSimpleName() + " Project";
+    }
+
+    private void doSetup()
+    {
+        _userHelper.createUser(PROJECT_CREATOR_USER, true, true);
+        _userHelper.createUser(READER, true, true);
+        new ApiPermissionsHelper(this).addMemberToRole(PROJECT_CREATOR_USER, "Project Creator", PermissionsHelper.MemberType.user, "/");
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+        _containerHelper.deleteProject(PROJECT_NAME_PC, afterTest, 1000000);
+        _userHelper.deleteUsers(false, PROJECT_CREATOR_USER, READER);
+    }
+
+    @Test
+    public void testCreateProjectByProjectCreator() throws IOException, CommandException
+    {
+        log("Project Creator creating the project with admin permission");
+        impersonate(PROJECT_CREATOR_USER);
+        Map<String, Object> params = new HashMap<>();
+        params.put("name", PROJECT_NAME_PC);
+        params.put("assignProjectAdmin", "true");
+        params.put("folderType", "Collaboration");
+        createProject(params);
+        stopImpersonating();
+
+        log("Verifying the permissions");
+        goToProjectHome(PROJECT_NAME_PC);
+        navBar().goToPermissionsPage().assertPermissionSetting(PROJECT_CREATOR_USER, "Project Administrator");
+        navBar().goToPermissionsPage().assertPermissionSetting(PROJECT_CREATOR_USER, "Folder Administrator");
+
+        log("Cleanup : Deleting the project");
+        _containerHelper.deleteProject(PROJECT_NAME_PC, false, WAIT_FOR_PAGE);
+
+        log("Project Creator creating the project without admin permission");
+        goToHome();
+        impersonate(PROJECT_CREATOR_USER);
+        params = new HashMap<>();
+        params.put("name", PROJECT_NAME_PC);
+        params.put("assignProjectAdmin", "false");
+        params.put("folderType", "Collaboration");
+        createProject(params);
+        stopImpersonating();
+
+        log("Verifying the permissions");
+        goToProjectHome(PROJECT_NAME_PC);
+        navBar().goToPermissionsPage().assertNoPermission(PROJECT_CREATOR_USER, "No Permissions");
+    }
+
+    @Test
+    public void testCreateProjectByReader() throws IOException, CommandException
+    {
+        log("Cleanup : Deleting the project");
+        _containerHelper.deleteProject(PROJECT_NAME_PC, false, WAIT_FOR_PAGE);
```

Comment:
should this be done in a @Before method since so that if more test cases are added in the future, they don't have to worry about the order of execution?